x-pack/legacy/plugins/apm/public/components/app/ServiceMap/Controls.tsx

```diff
@@ -86,45 +88,69 @@ export function Controls() {
   const minZoom = cy.minZoom();
   const isMinZoom = zoom === minZoom;
   const increment = (maxZoom - minZoom) / steps;
+
+  const centerLabel = i18n.translate('xpack.apm.serviceMap.center', {
+    defaultMessage: 'Center'
+  });
+  const viewFullMapLabel = i18n.translate('xpack.apm.serviceMap.viewFullMap', {
+    defaultMessage: 'View full service map'
+  });
   const zoomInLabel = i18n.translate('xpack.apm.serviceMap.zoomIn', {
     defaultMessage: 'Zoom in'
   });
   const zoomOutLabel = i18n.translate('xpack.apm.serviceMap.zoomOut', {
     defaultMessage: 'Zoom out'
   });
-  const centerLabel = i18n.translate('xpack.apm.serviceMap.center', {
-    defaultMessage: 'Center'
-  });
+
+  const showViewFullMapButton = cy.nodes('.primary').length > 0;
 
   return (
     <ControlsContainer>
-      <ZoomPanel hasShadow={true} paddingSize="none">
-        <ZoomInButton
-          aria-label={zoomInLabel}
-          color="text"
-          disabled={isMaxZoom}
-          iconType="plusInCircleFilled"
-          onClick={zoomIn}
-          title={zoomInLabel}
-        />
-        <Button
-          aria-label={zoomOutLabel}
-          color="text"
-          disabled={isMinZoom}
-          iconType="minusInCircleFilled"
-          onClick={zoomOut}
-          title={zoomOutLabel}
-        />
-      </ZoomPanel>
-      <EuiPanel hasShadow={true} paddingSize="none">
-        <Button
-          aria-label={centerLabel}
-          color="text"
-          iconType="crosshairs"
-          onClick={center}
-          title={centerLabel}
-        />
-      </EuiPanel>
+      <Panel hasShadow={true} paddingSize="none">
+        <EuiToolTip anchorClassName="eui-displayInline" content={zoomInLabel}>
+          <ZoomInButton
+            aria-label={zoomInLabel}
+            color="text"
+            disabled={isMaxZoom}
+            iconType="plusInCircleFilled"
+            onClick={zoomIn}
+          />
+        </EuiToolTip>
+        <EuiToolTip anchorClassName="eui-displayInline" content={zoomOutLabel}>
+          <Button
+            aria-label={zoomOutLabel}
+            color="text"
+            disabled={isMinZoom}
+            iconType="minusInCircleFilled"
+            onClick={zoomOut}
+          />
+        </EuiToolTip>
+      </Panel>
+      <Panel hasShadow={true} paddingSize="none">
+        <EuiToolTip anchorClassName="eui-displayInline" content={centerLabel}>
+          <Button
+            aria-label={centerLabel}
+            color="text"
+            iconType="crosshairs"
+            onClick={center}
+          />
+        </EuiToolTip>
+      </Panel>
+      {showViewFullMapButton && (
+        <Panel hasShadow={true} paddingSize="none">
+          <EuiToolTip
+            anchorClassName="eui-displayInline"
+            content={viewFullMapLabel}
+          >
+            <Button
+              aria-label={viewFullMapLabel}
+              color="text"
+              href={getAPMHref('/service-map', currentSearch)}
```

Comment:
we should pass all of the params (`useUrlParams().urlParams`) in the 3rd argument to `getAPMHref`, otherwise it will lose filters like date/time range and environment as they navigate.

Reply:
Updated in 9987f8ba12.